presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeAssignmentStats.java

```diff
@@ -39,12 +39,12 @@ public NodeAssignmentStats(NodeTaskMap nodeTaskMap, NodeMap nodeMap, List<Remote
         this.stageQueuedSplitInfo = new HashMap<>(nodeMapSize);
 
         for (RemoteTask task : existingTasks) {
-            checkArgument(stageQueuedSplitInfo.put(task.getNodeId(), new PendingSplitInfo(task.getQueuedPartitionedSplitCount())) == null, "A single stage may not have multiple tasks running on the same node");
+            checkArgument(stageQueuedSplitInfo.put(task.getNodeId(), new PendingSplitInfo(task.getQueuedPartitionedSplitCount(), task.getUnacknowledgedPartitionedSplitCount())) == null, "A single stage may not have multiple tasks running on the same node");
         }
 
         // pre-populate the assignment counts with zeros
         if (existingTasks.size() < nodeMapSize) {
-            Function<String, PendingSplitInfo> createEmptySplitInfo = (ignored) -> new PendingSplitInfo(0);
+            Function<String, PendingSplitInfo> createEmptySplitInfo = (ignored) -> new PendingSplitInfo(0, 0);
```

Comment:
Should we have a default constructor `PendingSplitInfo()` that set both values to 0?

Reply:
That's definitely an option. In this case I think it's more clear to say "we know that there is no task created for these nodes, so the values here are 0" than the "default constructor" necessarily signals, but I'm happy to make the change if you'd like.

Reply:
No worries. I don't have strong opinion on that